src/App.js

```diff
@@ -26,39 +28,26 @@ class App extends Component {
               <a href="#">Auth0 - React</a>
             </Navbar.Brand>
             <Button
-              bsStyle="primary"
-              className="btn-margin"
-              onClick={this.goTo.bind(this, 'home')}
-            >
-              Home
-            </Button>
-            {
-              !isAuthenticated() && (
-                  <Button
-                    bsStyle="primary"
-                    className="btn-margin"
-                    onClick={this.login.bind(this)}
-                  >
-                    Log In
-                  </Button>
-                )
-            }
-            {
-              isAuthenticated() && (
-                  <Button
-                    bsStyle="primary"
-                    className="btn-margin"
-                    onClick={this.logout.bind(this)}
-                  >
-                    Log Out
-                  </Button>
-                )
-            }
+              {...{
+                bsStyle: 'primary',
+                className: 'btn-margin',
+                onClick: () => this.goTo.bind(this, 'home'),
+                value: 'Home'
+              }}
+            />
+            <Button
+              {...{
+                bsStyle: 'primary',
+                className: 'btn-margin',
+                onClick: () => (authenticated ? this.logout.bind(this) : this.login.bind(this)),
```

Comment:
I changed your conditionals to be in a single shared button where we just change the props we need. I would generally prefer this logic be separated out into it's own function as well, but I took the easy route here so as to not change too much. 

Ideally, I'd prefer to see something like this: 



```
const handleClick = () => {
  const { isAuthenticated } = this.props
  return authenticated ? this.logout.bind(this) : this.login.bind(this)
}

...

<Button 
  {...{
    onClick: handleClick
  }}
/>
```